src/module_wrap.cc

```diff
@@ -208,12 +226,61 @@ void ModuleWrap::Instantiate(const FunctionCallbackInfo<Value>& args) {
 }
 
 void ModuleWrap::Evaluate(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
   Isolate* isolate = args.GetIsolate();
-  Local<Object> that = args.This();
-  Local<Context> context = that->CreationContext();
-  ModuleWrap* obj = Unwrap<ModuleWrap>(that);
-  CHECK_NE(obj, nullptr);
-  MaybeLocal<Value> result = obj->module_.Get(isolate)->Evaluate(context);
+  ModuleWrap* obj;
+  ASSIGN_OR_RETURN_UNWRAP(&obj, args.This());
+  Local<Context> context = obj->context_.Get(isolate);
+  Local<Module> module = obj->module_.Get(isolate);
+
+  Environment::ShouldNotAbortOnUncaughtScope no_abort_scope(env);
+  TryCatch try_catch(isolate);
+  Maybe<int64_t> maybe_timeout = GetTimeoutArg(env, args[0]);
```

Comment:
should timeout/sigint code be part of the default implementation of all modules. I feel like this is starting to tread ModuleWrap as strictly a JS consumable rather than being separated.